server/dive_tasks/tasks.py

```diff
@@ -545,3 +545,59 @@ def convert_images(self: Task, folderId):
             str(folderId),
             {"annotate": True},  # mark the parent folder as able to annotate.
         )
+
+
+@app.task(bind=True, acks_late=True, ignore_result=True)
+def extract_zip(self: Task, folderId: str, itemId: str):
+    context: dict = {}
+    gc: GirderClient = self.girder_client
+    manager: JobManager = patch_manager(self.job_manager)
+    if utils.check_canceled(self, context):
+        manager.updateStatus(JobStatus.CANCELED)
+        return
+
+    with tempfile.TemporaryDirectory() as _working_directory, suppress(utils.CanceledError):
+        _working_directory_path = Path(_working_directory)
+        item: GirderModel = gc.getItem(itemId)
+        file_name = str(_working_directory_path / item['name'])
+        manager.write(f'Fetching input from {itemId} to {file_name}...\n')
+        gc.downloadItem(itemId, _working_directory, item["name"])
+        with zipfile.ZipFile(file_name, 'r') as zipObj:
+            listOfFileNames = zipObj.namelist()
+            sum_file_size = sum([data.file_size for data in zipObj.filelist])
+            sum_compress_size = sum([data.compress_size for data in zipObj.filelist])
+            ratio = sum_file_size / sum_compress_size
+            if ratio > 600:
+                manager.write(
+                    f"Compression ratio is exceedingly high at {ratio}\n\
+                    Please contact an admin at viame-web@kitware.com if this is a valid zip file"
+                )
+                manager.updateStatus(JobStatus.ERROR)
+                return
+
+            for fileName in listOfFileNames:
+                if fileName.endswith('.zip'):
+                    manager.write("Nested Zip files are invalid!!!!\n")
+                    manager.updateStatus(JobStatus.ERROR)
+                    return
+                manager.write(f"Extracting: {fileName}\n")
+                zipObj.extract(fileName, f'{_working_directory}/{fileName}')
+        # validation of files in folder using dive/data
+        validation = gc.sendRestRequest(
+            'POST', '/dive_dataset/validate_files', json=listOfFileNames
+        )
+        if 'ok' in validation.keys() and validation['ok'] is True:
```

Comment:
```suggestion
        if validation.get('ok', False):
```

Python has a lot of features that are just _better_ than JS, and object.get('key', default) is one of them.